tsl/src/compression/compression.c

```diff
@@ -67,6 +67,41 @@ static const CompressionAlgorithmDefinition definitions[_END_COMPRESSION_ALGORIT
 	[COMPRESSION_ALGORITHM_DELTADELTA] = DELTA_DELTA_ALGORITHM_DEFINITION,
 };
 
+#if PG14_GE
+/* The prefix of a logical replication message which is inserted into the
+ * replication stream right before decompression inserts are happening
+ */
+#define DECOMPRESSION_MARKER_START "::timescaledb-decompression-start"
+/* The prefix of a logical replication message which is inserted into the
+ * replication stream right after all decompression inserts have finished
+ */
+#define DECOMPRESSION_MARKER_END "::timescaledb-decompression-end"
+
+static inline void
+write_logical_replication_msg_decompression_start()
+{
+	if (XLogLogicalInfoActive())
+	{
+		DirectFunctionCall3(pg_logical_emit_message_text,
```

Comment:
Why don't use the internal Postgres API "LogLogicalMessage" instead?

Reply:
Didn't know about the function, I was happy to actually make it work "somehow" 😅